test/parallel/test-async-hooks-http-parser-destroy.js

```diff
@@ -59,3 +53,22 @@ server.listen(0, function() {
     })();
   }
 });
+
+function checkOnExit() {
+  assert.deepStrictEqual(destroyedIds.sort(), createdIds.sort());
+  // Each HTTP request results in two createdIds and two destroyedIds.
+  assert.strictEqual(createdIds.length, N * 2);
+}
+
+// Ordinary exit.
+process.on('exit', checkOnExit);
+
+// Test runner tools/test.py will send SIGTERM if timeout.
+// Signals aren't received in workers, but that's OK. The test will still fail
+// if it times out. It just won't provide the additional information that this
+// handler does.
+process.on('SIGTERM', () => {
+  console.error('Received SIGTERM. (Timed out?)');
+  checkOnExit();
+  process.exit(1);
```

Comment:
Do you need `checkOnExit()` here if you already call `process.exit()`?

Reply:
I suppose not!